sql/2022/jamstack/candidate_urls.sql

```diff
@@ -0,0 +1,88 @@
+-- step 1.1: get URLs with age headers
+with headers as (
+  SELECT 
+    url,
+    JSON_EXTRACT_ARRAY(payload, '$.response.headers') as headers_array
+  FROM `httparchive.requests.2022_07_01_mobile` 
+),
+
+flattened_headers as (
+  SELECT 
+    url,
+    LOWER(JSON_VALUE(flattened_headers,'$.name')) as header_name,
+    JSON_VALUE(flattened_headers,'$.value') as header_value
+  FROM headers
+  CROSS JOIN UNNEST(headers.headers_array) AS flattened_headers
+),
+
+non_null_ages as (
+  select 
+    url,
+    SAFE_CAST(header_value as numeric) as age 
+  from flattened_headers 
+  where header_name = 'age'
+  and SAFE_CAST(header_value as numeric) is not null
+),
+
+-- step 1.2: filter URLs to age headers at our chosen level
+age_filtered as (
+  select 
+    distinct(url) as url
+  from non_null_ages
+  where age > 86400 -- 1 day
+),
+
+-- step 2.1: get URLs with LCP times
+-- step 3.2: get URLs with CLS times
+lighthouse_audits as (
+  SELECT  
+    url,
+    CAST(JSON_EXTRACT(report, '$.audits.largest-contentful-paint.numericValue') as NUMERIC) as lcp_ms,
+    CAST(JSON_EXTRACT(report, '$.audits.cumulative-layout-shift.numericValue') as NUMERIC) as cls

```

Comment:
We should not use the lighthouse data for this. Instead either use the CrUX data (in the `pages` data set) if wanting home page data, or the CrUX data in `chrome-ux-report.materialized.device_summary` dataset if wanting to use the whole origin data.